app/code/Magento/InventoryInStorePickup/etc/di.xml

```diff
@@ -35,4 +35,15 @@
             <argument name="identityContainer" xsi:type="object">\Magento\InventoryInStorePickup\Model\Order\Email\Container\ReadyForPickupIdentity</argument>
         </arguments>
     </type>
+    <type name="Magento\InventoryInStorePickup\Model\SourceSelection\GetActiveStorePickupOrdersBySource">
+        <arguments>
+            <argument name="statesToFilter" xsi:type="array">
+                <item xsi:type="const" name="complete">Magento\Sales\Model\Order::STATE_COMPLETE</item>
+                <item xsi:type="const" name="closed">Magento\Sales\Model\Order::STATE_CLOSED</item>
+                <item xsi:type="const" name="canceled">Magento\Sales\Model\Order::STATE_CANCELED</item>
+            </argument>
+        </arguments>
+    </type>
+    <preference for="Magento\InventorySourceSelectionApi\Model\GetSourceItemQtyAvailableInterface"
```

Comment:
How do we handle situation where Stock Pickup is not enabled or prioritization isn't necessary?

Reply:
As a variant, by implementing the TODO above.
I feel that replacing the original functionality is not the best solution - so I can move it from preference to a plugin.

Reply:
Already discussed - see the comment above.